test/coinselector-test.js

```diff
@@ -0,0 +1,335 @@
+        }
+
+        assert.strictEqual(selectedValues, target);
+        assert(lowFeeSelection.length > highFeeSelection.length);
+      });
+    }
+  });
+
+  describe('Lowest Larger Selection', function () {
+    // try selecting a single UTXO
+    for (const value of values) {
+      it(`should select target=${value} using Lowest Larger`, () => {
+        const selection = selector.selectLowestLarger(value);
+        assert.strictEqual(selection.length, 1);
+        assert.strictEqual(pointers[selection[0]].effectiveValue, value);
+      });
+    }
+
+    // these targets may or may not have exact solutions
+    for (const target of targetSet1) {
+      it(`should select target=${target} using Lowest Larger`, () => {
+        const selection = selector.selectLowestLarger(target);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert(selectedValues >= target);
+      });
+    }
+
+    // Lowest Larger should select (target - 500)
+    for (const target of targetSet1) {
+      it(`should select target=${target - 500} using Lowest Larger`, () => {
+        const selection = selector.selectLowestLarger(target - 500);
+        assert(selection.length > 0);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert(selectedValues > target - 500);
+      });
+    }
+
+    // these targets have exact solution
+    for (const target of targetSet3) {
+      it(`should select target=${target} using Lowest Larger for exact matches`, () => {
+        const selection = selector.selectLowestLarger(target);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert.strictEqual(selectedValues, target);
+      });
+    }
+
+    it('should be able to fund all values in range 1 to 221000 using Lowest Larger', () => {
+      for (let target = 1; target <= 221000; target++) {
+        const selection = selector.selectLowestLarger(target);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert(selectedValues >= target);
+      }
+    });
+  });
+
+  describe('Single Random Draw Selection', function () {
+    it('should be able to fund all values in range 1 to 221000 using Single Random Draw', () => {
+      for (let target = 1; target <= 221000; target++) {
+        const selection = selector.selectSRD(target);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert(selectedValues >= target);
+      }
+    });
+  });
+});
+
+const workers = new WorkerPool({
+  enabled: true,
+  size: 2
+});
+
+describe('Integration', function () {
+  this.timeout(1000000);
```

Comment:
Does this have to be ~16 minutes? The test runs in about 3 minutes on my laptop, just over 4 minutes on CI. Let's bring it down to a reasonable level (5 maybe 6 minutes)